### DISCUSSION TITLE
Why does AsFailedTask use TaskCompletionSource instead of Task FromException

### DISCUSSION THREAD

**Discussion by zms9110750** [louthy/language-ext#1353] [Q&A]
```csharp
	public static Task<A> AsFailedTask<A>(this Exception ex)
	{
		return Task.FromException<A>(ex); 
	}
```

**Comment by louthy** [comment]
I don't remember entirely, but I think that FromException didn't exist in early .netcore/netstandard.

**Reply by zms9110750** [reply]
https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.fromexception?view=net-8.0
Core 1.0 , Framework 4.6 , Standard 1.3.

**Reply by zms9110750** [reply]
TaskCompletionSource was first introduced on. net5.
https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskcompletionsource?view=net-8.0
